controllers/druid/drain_historical.go

```diff
@@ -0,0 +1,191 @@
+package druid
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	//TODO cleanup imports
+
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger_drain_historical = logf.Log.WithName("drain_historical")
+
+type CoordinatorConfig struct {
+	decommissioningNodes []string
+}
+
+func checkDrainStatus(d DruidClient, start int32, end int32, podNames []string) (bool, error) {
+	historicalUsageStats, err := d.GetHistoricalUsage()
+	if err != nil {
+		return false, err
+	}
+	for it := start; it <= end; it++ {
+		if historicalUsageStats[podNames[it]] != 0 {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func patchUpdateStrategy(sdk client.Client, m *v1alpha1.Druid, nodeSpec *v1alpha1.DruidNodeSpec, strategy string, emitEvent EventEmitter) error {
+	updatedObj := nodeSpec.DeepCopy()
+
+	newStrategy := appsv1.StatefulSetUpdateStrategy{
+		Type: appsv1.OnDeleteStatefulSetStrategyType,
+	}
+
+	updatedObj.UpdateStrategy = &newStrategy
+
+	if err := writers.Patch(context.TODO(), sdk, m, updatedObj, false, client.MergeFrom(nodeSpec), emitEvent); err != nil {
+		return err
+	} else {
+		msg := fmt.Sprintf("Successfully Patched %s with %s", nodeSpec.UpdateStrategy, onDelete)
+		logger_drain_historical.Info(msg, "name", m.Name, "namespace", m.Namespace)
+		return nil
+	}
+}
+
+func scaleStatefulSet(sdk client.Client, m *v1alpha1.Druid, nodeSpec *v1alpha1.DruidNodeSpec, strategy string, emitEvent EventEmitter, scaleUpCount int32) error {
+	updatedObj := nodeSpec.DeepCopy()
+	newReplica := int32(m.Status.Historical.Replica + scaleUpCount)
+	updatedObj.Replicas = newReplica
+	if err := writers.Patch(context.TODO(), sdk, m, updatedObj, false, client.MergeFrom(nodeSpec), emitEvent); err != nil {
+		return err
+	} else {
+		msg := fmt.Sprintf("Successfully Patched replicas with %d", newReplica)
+		logger_drain_historical.Info(msg, "name", m.Name, "namespace", m.Namespace)
+		return nil
+	}
+}
+
+func deployHistorical(sdk client.Client, m *v1alpha1.Druid, nodeSpec *v1alpha1.DruidNodeSpec, nodeSpecUniqueStr string, emitEvent EventEmitter, batchSize int32, baseUrl string) error {
+	// patch the updateStrategy with onDelete
+	err := patchUpdateStrategy(sdk, m, nodeSpec, onDelete, emitEvent)
```

Comment:
This means we would be patching the update strategy on every run. Can we patch it only at the start of an deployment process? Also, make sure to revert the patching done here after the deployment.